packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -109,6 +137,46 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         emit FlashWMint(msg.sender, _vaultId, _wPowerPerpAmount, amountToFlashswap, _collateralAmount);
     }
 
+    /**
+     * @notice flash close position and buy long squeeth
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmount amount of WPowerPerp to burn
+     * @param _collateralToWithdraw amount of collateral to withdraw
+     * @param _minToReceive minimum amount of long WPowerPerp to receive
+     */
+    function flashWBurnBuyLong(
```

Comment:
maybe need to tweak names as well:

(if changed based on my comments) this is basically a function that can:
1. buyback and fully close a short
2. go long with any amount >= 0 wPowerPerp

So maybe fullCloseAndLong, closeShortAndBuy?